tools/benchmarks/shared/src/main/kotlin/report/BenchmarksReport.kt

```diff
@@ -30,50 +26,62 @@ interface JsonSerializable {
 }
 
-// Class for benchcmarks report with all information of run.
-class BenchmarksReport(val env: Environment, benchmarksList: List<BenchmarkResult>, val compiler: Compiler):
+// Parse array with benchmarks to list
+fun parseBenchmarksArray(data: JsonElement): List<BenchmarkResult> {
+    if (data is JsonArray) {
+        return data.jsonArray.map {
+            if (MeanVarianceBenchmark.isMeanVarianceBenchmark(it))
+                MeanVarianceBenchmark.create(it as JsonObject)
+            else BenchmarkResult.create(it as JsonObject)
+        }
+    } else {
+        error("Benchmarks field is expected to be an array. Please, check origin files.")
+    }
+}
+
+// Class for benchmarks report with all information of run.
+open class BenchmarksReport(val env: Environment, benchmarksList: List<BenchmarkResult>, val compiler: Compiler) :
         JsonSerializable {
 
-    companion object: EntityFromJsonFactory<BenchmarksReport> {
+    companion object : EntityFromJsonFactory<BenchmarksReport> {
         override fun create(data: JsonElement): BenchmarksReport {
             if (data is JsonObject) {
                 val env = Environment.create(data.getRequiredField("env"))
                 val benchmarksObj = data.getRequiredField("benchmarks")
                 val compiler = Compiler.create(data.getRequiredField("kotlin"))
+                val buildNumberField = data.getOptionalField("buildNumber")
                 val benchmarksList = parseBenchmarksArray(benchmarksObj)
-                return BenchmarksReport(env, benchmarksList, compiler)
+                val report = BenchmarksReport(env, benchmarksList, compiler)
+                buildNumberField?.let { report.buildNumber = (it as JsonLiteral).unquoted() }
+                return report
             } else {
                 error("Top level entity is expected to be an object. Please, check origin files.")
             }
         }
 
-        // Parse array with benchmarks to list
-        fun parseBenchmarksArray(data: JsonElement): List<BenchmarkResult> {
-            if (data is JsonArray) {
-                return data.jsonArray.map { BenchmarkResult.create(it as JsonObject) }
-            } else {
-                error("Benchmarks field is expected to be an array. Please, check origin files.")
-            }
-        }
-
         // Made a map of becnhmarks with name as key from list.
         private fun structBenchmarks(benchmarksList: List<BenchmarkResult>) =
-                benchmarksList.groupBy{ it.name }
+                benchmarksList.groupBy { it.name }
     }
 
-    val benchmarks: Map<String, List<BenchmarkResult>> = structBenchmarks(benchmarksList)
+    val benchmarks = structBenchmarks(benchmarksList)
 
-    override fun toJson(): String {
+    var buildNumber: String? = null
+
+    override fun serializeFields(): String {
+        val buildNumberField = buildNumber?.let {
+            """,
+            "buildNumber": "$buildNumber"
+        """
```

Comment:
A broken indent here. I see that methods like `trimIndent()` are not used while writing multiline strings, so the resulting JSON would be hard to read due to inconsistant indents.